aspnetcore/mvc/controllers/actions.md

```diff
@@ -14,80 +14,91 @@ uid: mvc/controllers/actions
+Public methods on a controller, except those decorated with the `[NonAction]` attribute, are actions. Parameters on actions are bound to request data and are validated using [model binding](xref:mvc/models/model-binding). Model validation occurs for everything that's model-bound. The `ModelState.IsValid` property value indicates whether model binding and validation succeeded.
 
->[!WARNING]
-> Action methods that accept parameters should verify the `ModelState.IsValid` property is true.
+Action methods should contain logic for mapping a request to a business concern. Business concerns should typically be represented as services that the controller accesses through [dependency injection](xref:mvc/controllers/dependency-injection). Actions then map the result of the business action to an application state.
 
-Action methods should contain logic for mapping an incoming request to a business concern. Business concerns should typically be represented as services that your controller accesses through [dependency injection](dependency-injection.md). Actions then map the result of the business action to an application state.
-Actions can return anything, but frequently will return an instance of `IActionResult` (or `Task<IActionResult>` for async methods) that produces a response. The action method is responsible for choosing *what kind of response*; the action result *does the responding*.
+Actions can return anything, but frequently return an instance of `IActionResult` (or `Task<IActionResult>` for async methods) that produces a response. The action method is responsible for choosing *what kind of response*. The action result *does the responding*.
 
 ### Controller Helper Methods
 
-Although not required, most developers will want to have their controllers inherit from the base `Controller` class. Doing so provides controllers with access to many properties and helpful methods, including the following helper methods designed to assist in returning various responses:
+Controllers usually inherit from [Controller](https://docs.microsoft.com/aspnet/core/api/microsoft.aspnetcore.mvc.controller), although this is not required. Deriving from `Controller` provides access to three categories of helper methods:
 
-**[View](../views/index.md)**
+#### 1. Methods resulting in an empty response body
 
-Returns a view that uses a model to render HTML. Example: `return View(customer);`
+No `Content-Type` HTTP response header is included, since the response body lacks content to describe.
 
-**HTTP Status Code**
+There are two result types within this category: Redirect and HTTP Status Code.
 
-Return an HTTP status code. Example: `return BadRequest();`
+* **HTTP Status Code**
 
-**Formatted Response**
+    This type returns an HTTP status code. A couple helper methods of this type are `BadRequest`, `NotFound`, and `Ok`. For example, `return BadRequest();` produces a 400 status code when executed.
```

Comment:
Should clarify that `BadRequest` and so on fit in this category when called without parameters.